### ISSUE TITLE
add `no-unknown-schema` rule

### ISSUE THREAD

**Issue by marcalexiei** [marcalexiei/eslint-plugin-zod#34]
Enforce disallowing `z.unknown()` usage entirely, pushing you to use more explicit schemas.

### Options

N/A

### Configuration 

```ts
export default defineConfig({
  // ...
  rules: {
   'zod/no-unknown-schema': ['error']
  }
  // ...
})
```

### Invalid

```ts
const stringSchema = z.unknown();
```

### Valid

```ts
const stringSchema = z.string().refine(() => true, {
  error: `asd`,
});
```

Add a mentation of this rule in `no-any` documentation since it might cause conflicts with suggestion from that rule
